packages/api-clients/admin-api-client/src/validations.ts

```diff
@@ -16,8 +16,8 @@ export function validateRequiredAccessToken(accessToken: string) {
   }
 }
 
-export function validateServerSideUsage() {
-  if (typeof window !== 'undefined') {
+export function validateServerSideUsage(isTesting = false) {
+  if (typeof window !== 'undefined' && !isTesting) {
```

Comment:
This was required to be able to unit test using JSDOM.

Reply:
I also considered an environmental variable set by something like `shopify app testing`, but that was a little too complicated for me to coordinate across repositories and implement, and it would also be similar in implementation to a flag set on config. Detecting JSDOM is a more brittle solution--I would recommend against this approach.

Reply:
Totally agree on detecting jsdom, probably not a good option. I think we can keep this for now and consider some way to not need any flags in the future.

Reply:
This is a risk based decision in my view, so I'll be guided by your assessment on "what's the worst that could happen?". Throwing an error prevents it from being used in a browser context, but `console.error` only warns the developer.